### DISCUSSION TITLE
Textnode components are disappearing once style is applied

### DISCUSSION THREAD

**Discussion by alec-braze** [GrapesJS/grapesjs#5062] [Q&A] [answered]
grapes version: 0.20.4

I'm having an issue where `textnode` components are getting dropped from my `text` component when styling is applied to multiple `textnode`s independently.

Steps to reproduce (in our custom editor): 
* Add a text component to the editor
* Enter some text
* Insert a newline
* Enter some more text
* Style the bottom text as bold, italic, etc.
* Style the top text as bold, italic, etc
* click outside the text components

Behavior: The bottom text disappears on blur

Notably this doesn't happen if the styles are applied simultaneously. If I watch the DOM in the devtools I can see the top text getting turned into a custom component, but the bottom text doesn't, which I think is why it's dropping. 

We have a custom component for handling stylized text

```import grapesjs from "grapesjs";
import { COMPONENT_TYPE_GRAPES_TEXT } from "./TextComponentPlugin";

export const COMPONENT_TYPE_STYLED_TEXT = "styled-text";
const styledTextTags = ["i", "u", "b", "s", "strike"];

export const StyledTextComponentPlugin = (editor: grapesjs.Editor) => {
  editor.DomComponents.addType(COMPONENT_TYPE_STYLED_TEXT, {
    extend: COMPONENT_TYPE_GRAPES_TEXT,
    isComponent: (el: HTMLElement) => {
      const isStyledText = styledTextTags.includes(el.tagName?.toLowerCase());
      if (isStyledText) {
        return {
          ...editor.DomComponents.getType(COMPONENT_TYPE_STYLED_TEXT),
          type: COMPONENT_TYPE_GRAPES_TEXT,
        };
      }
    },
  });
};
```

and we've slightly modified the builtin text component

```
export const TextComponentPlugin = (editor: grapesjs.Editor, { t }: PluginOpts) => {
  editor.DomComponents.addType(COMPONENT_TYPE_GRAPES_TEXT, {
    extend: COMPONENT_TYPE_GRAPES_TEXT,
    model: {
      defaults: {
        hoverable: false,
        highlightable: false,
        draggable: false,
      },
    },
    view: {
      events: {
        events: {
          click: "clickHandler",
        },
      },
      clickHandler() {
        // @ts-ignore
        this.onActive();
      },
    },
  });
};
```

But I don't see why, if single `textnodes` correctly convert to `text`, or multiple `textnodes` correctly convert when styled together, styling them separately causes this. 
The closest similar issue I could find is https://github.com/GrapesJS/grapesjs/issues/3012 , but my issue isn't happening when saving and re-opening the editor, it's happening on blur. 

This doesn't happen in [the grapes demo](https://grapesjs.com/demo.html) so it must be something specific to how we're handling text?

**Comment by alec-braze** [comment]
![text_styling](https://user-images.githubusercontent.com/116038925/233443920-c942174d-e5db-495d-8b0f-6b2b4581b67d.gif)
gif of it happening, including the dev console so you can see what's happening in the DOM

**Comment by artf** [comment, marked as answer]
Hi @alec-braze, what is the point of `...editor.DomComponents.getType(COMPONENT_TYPE_STYLED_TEXT)` in COMPONENT_TYPE_STYLED_TEXT component?
`getType` returns the entire component definition (model, view, etc.) not sure that is the real intention there and looks like the root of the issue

**Comment by alec-braze** [comment]
@artf awesome, removing that line did indeed fix it! I'm not sure what the original intention of that line was, I'll check with the engineer that wrote it, but I can definitely see in retrospect why that would be a problem, and why it was so frustrating to debug! My understanding is that having that line in there probably made it so that the extra styled lines were having their content erased because `isComponent` wasn't just returning a truthy/falsy value, but a lot of other things, and probably overriding some necessary values. I really appreciate the help! 

Since it currently works I'm going to take it one step further and see if we even technically need the StyledTextComponentPlugin at all.
